File: src/main/java/ru/tykvin/hermes/file/model/UploadingEntity.java
package ru.tykvin.hermes.file.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.apache.tomcat.util.http.fileupload.FileItemStream;

import java.time.OffsetDateTime;
import java.util.UUID;

@Builder
@Getter
@Setter
public class UploadingEntity {
    private final FileItemStream item;
    private final UUID id = UUID.randomUUID();
    private final OffsetDateTime creatAt = OffsetDateTime.now();
    private long size;
    private String fileName;
    private String sha256;
    private boolean uploaded;
    private boolean tmp;
}

File: src/main/kotlin/ru/tykvin/hermes/file/service/dao/UploadingEntity.kt
package ru.tykvin.hermes.file.service.dao

import org.apache.tomcat.util.http.fileupload.FileItemStream
import java.time.OffsetDateTime
import java.util.*

class UploadingEntity(
        val item: FileItemStream,
        val id: UUID = UUID.randomUUID(),
        val creatAt: OffsetDateTime = OffsetDateTime.now(),
        val size: Long = 0,
        val fileName: String,
        val sha256: String,
        val uploaded: Boolean = false,
        val tmp: Boolean = true
)
